api/v1beta2/conditions_consts.go

```diff
@@ -120,4 +120,8 @@ const (
 	// CreateInfrastructureAnnotation is the name of an annotation that indicates if
 	// Power VS infrastructure should be created as a part of cluster creation.
 	CreateInfrastructureAnnotation = "powervs.cluster.x-k8s.io/create-infra"
+
+	// CreateVPCInfrastructureAnnotation is the name of an annotation that indicates if
+	// VPC infrastructure should be created as part of cluster creation.
+	CreateVPCInfrastructureAnnotation = "vpc.cluster.x-k8s.io/create-infra"
```

Comment:
@cjschaef what will be used to differentiate v2 vs v1 code path. I think in VPC we create infra by default so this annotation might not be required. Are you planning to use this as a differentiator?

Reply:
Yes, I plan on using that and flagging off the NetworkSpec field in the Cluster resource for the v2 path.

Reply:
ok, If I understand correctly, in V2 path you are expecting `NetworkSpec` to be mandatory?

Reply:
That is correct. `NetworkSpec` is part of the new Infrastructure deployment support, and is required for that support.

Reply:
Got it thanks for the clarification, 
May be we can get started with using that annotation if we can't find any alternative suitable name which matches for what we are doing but eventually we need to remove it as it may confuse the users because now also by default we create the resource.

Reply:
I could remove the annotation altogether, if we wish to start the so-called "v2" VPC support without requiring an annotation (rely only on using `NetworkSpec` to flag the new (v2) VPC support, creating all of the required Infrastructure.

I had expected that annotation was required in order to get CAPI to perform Infrastructure Creation/Management. But if we expect for our new (v2) support, CAPI will always be attempting to create/manage Infrastructure resources, I can drop that annotation completely.

Reply:
Another question, can CAPI be run within a cluster with only the Machine Controller running (not the Cluster/Infra Controller), so as to not attempt to reprovision Infrastructure post cluster creation/install? I had expected this annotation would help with that situation as well, to prevent additional churn by CAPI by continuing to monitor (or deleting) Infrastructure post cluster create.

Reply:
In PowerVS v1 flow we were not creating any infra rather using existing infra and expecting user to provide them. We introduced this annotation in for v2 to make sure we need to create infra and not expecting anything from user.

This is not true for VPC, in vpc v1 flow as well we create required resources so this annotation does imply anything. But for you to differentiate V2 path as you suggested its better to go with NetworkSpec. If user set NetworkSpec you can assume its V2 flow.

Reply:
>Another question, can CAPI be run within a cluster with only the Machine Controller running (not the Cluster/Infra Controller), so as to not attempt to reprovision Infrastructure post cluster creation/install? I had expected this annotation would help with that situation as well, to prevent additional churn by CAPI by continuing to monitor (or deleting) Infrastructure post cluster create.

No this annotation is nothing to do with that, But to achieve what you are expecting capi provides annotation named `[cluster.x-k8s.io/managed-by](https://github.com/Karthik-K-N/cluster-api/blob/e2bf2b7d108ecd51bf7b23bf50e2432e82d8f2fe/api/v1beta1/common_types.go#L154-L160)` and our controllers are configured to use them [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/01e1771ca8cbf8b9969bb64b6956b3d470219313/controllers/ibmpowervscluster_controller.go#L477).

Reply:
Okay, thanks for the details, I will drop the annotation. and rely only on the NetworkSpec.